### PR TITLE
Add a "debug" config to print HTTP requests and responses

### PR DESCRIPTION
**Pull request by Nyholm**
It could be helpful to see exactly what HTTP requests and responses we are dealing with. The official SDK prints this to stdout but we use PSR3.

### CODE REVIEW

File: src/Core/src/AbstractApi.php [hunk 1]
```diff
@@ -136,6 +136,15 @@ protected function getSignatureScopeName(): string
             ]
         );
 
+        if (\filter_var($configuration->get('debug'), \FILTER_VALIDATE_BOOLEAN)) {
+            $this->logger->debug('AsyncAws HTTP request sent', [
+                'method' => $request->getMethod(),
+                'endpoint' => $request->getEndpoint(),
+                'headers' => json_encode($request->getHeaders()),
+                'body' => 0 === $length ? null : $requestBody,
```
Comment (jderusse) [inline src/Core/src/AbstractApi.php:144]: is it an issue when body is a stream ?

**Review by jderusse — APPROVED** [review]
couldn't it be provided by the HttpClientLogger ? just asking

File: src/Core/src/AbstractApi.php (hunk 1 above)
Comment (Nyholm) [reply src/Core/src/AbstractApi.php:144]: We've converted the body to a string just a few lines above.

**Comment by Nyholm** [comment]
What is the `HttpClientLogger`? I've never heard of it and I cannot find any references to it..

**Comment by jderusse** [comment]
nvm, I thought Symfony provides such things. I was trapped by https://github.com/symfony/symfony/pull/30537/files

**Comment by Nyholm** [comment]
So after reading the source: The logger in the HttpClient does not capture any headers or body. I think that is important to include here.

**Comment by Nyholm** [comment]
This PR is ready for review

**Comment by jderusse** [comment]
thank you

**Pull request merged**